provider/datagen/src/lib.rs

```diff
@@ -308,9 +386,12 @@ pub fn keys_from_file<P: AsRef<Path>>(path: P) -> std::io::Result<Vec<DataKey>>
         .collect()
 }
 
-/// Parses a compiled binary and returns a list of [`DataKey`]s used by it.
+/// Parses a compiled binary and returns a list of [`DataKey`]s that it uses *at runtime*.
 ///
-/// Unknown key names are ignored.
+/// This function is intended to be used for binaries that use `AnyProvider` or `BufferProvider`,
+/// for which the compiler cannot remove unused data. Using this function on a binary that only
+/// uses compiled data (through the `compiled_data` feature or manual baked data) will not return
+/// any keys, as the keys only exist in the type system.
 //  Supports the hello world key
```

Comment:
Nit: Clean up this hello world aside while you're in here

Reply:
it's not a doc comment

Reply:
If it's not a doc comment it should ideally not be in the middle of a doc comment